piptools/repositories/pypi.py

```diff
@@ -331,6 +342,35 @@ def _wheel_support_index_min(self, tags=None):
             Wheel.support_index_min = original_support_index_min
             self._available_candidates_cache = original_cache
 
+    def _setup_logging(self):
+        """
+        Setup pip's logger. Ensure pip is verbose same as pip-tools and sync
+        pip's log stream with LogContext.stream.
+        """
+        # Default pip's logger is noisy, so decrease it's verbosity
+        setup_logging(
+            verbosity=log.verbosity - 1,
+            no_color=self.options.no_color,
+            user_log_file=self.options.log,
+        )
+
+        # Sync pip's console handler stream with LogContext.stream
+        logger = logging.getLogger()
+        console_handlers = [
+            handler for handler in logger.handlers if handler.name == "console"
+        ]
+        if console_handlers:  # pragma: no branch
+            console_handlers[0].stream = log.stream
+        else:  # pragma: no cover
+            # There is always a console handler. This warning would be a signal that
+            # this block should be removed/revisited, because of pip possibly
+            # refactored-out logging config.
+            log.warning("Couldn't find a 'console' logging handler")
```

Comment:
@AndydeCleyre 

A tiny update, I refactored this block for two reasons:
1. fix decreased code coverage
2. to be sure that we won't miss the moment when the console logger is gone, so we can remove or refactor this block.

What do you think?

Reply:
Sure, looks ok with clear comments. Adding a warning to track this is a good idea.

This may be better for coverage, but for goal 2, adding an `else` to the former `for` loop would've been a nice place for the warning, maintaining that `break` upon finding the right logger. 

Alternatively, and this may be a less readable alternative, you could make that a generator instead of a comprehension, `next` it to get the first (only) item, and put the warning under an `except StopIteration`.

Reply:
Yes, there are so many ways to do the same (hello python zen №13 😄), but I've chosen the current one deliberately as a more readable IMO. In fact, first version was with `for/else`, but this construction always confuses me.

Reply:
But I don't want to hold on this, if you feel strongly that the other ways are more suitable, I'd be happy to refactor this block and go-ahead 👍

Reply:
No problem, my approval stands for your current implementation. I only felt obligated to note out loud that this loops through handlers even after finding the right one. But that's not a real problem.

Looking forward to the merge! 👍

Reply:
I've reconsidered my opinion about `for-else` statement, hence #1117. Thank you @AndydeCleyre.